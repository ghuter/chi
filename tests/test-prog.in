import libc

main :: fun (argc: i32, argv: []u8) i32 {
	if argc == 2 {
		puts(argv[1])
	}

	else if argc == 2 {
		puts(argv[1])
		puts(argv[2])
	}

	else {
		puts("Hello, world")
	}

	return 0
}
