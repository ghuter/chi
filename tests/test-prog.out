IMPORT(60)
SPC(6)
IDENTIFIER(3)
NEWLINE(4)
NEWLINE(4)
IDENTIFIER(3)
SPC(6)
COLON(7)
COLON(7)
SPC(6)
LPAREN(10)
IDENTIFIER(3)
COLON(7)
SPC(6)
IDENTIFIER(3)
COMMA(8)
SPC(6)
IDENTIFIER(3)
COLON(7)
SPC(6)
LBRACKETS(12)
RBRACKETS(13)
IDENTIFIER(3)
RPAREN(11)
SPC(6)
IDENTIFIER(3)
SPC(6)
LBRACES(14)
NEWLINE(4)
SPC(6)
IF(53)
SPC(6)
IDENTIFIER(3)
SPC(6)
EQUAL(27)
SPC(6)
IMMEDIATE(2)
SPC(6)
LBRACES(14)
NEWLINE(4)
SPC(6)
IDENTIFIER(3)
LPAREN(10)
IDENTIFIER(3)
LBRACKETS(12)
IMMEDIATE(2)
RBRACKETS(13)
RPAREN(11)
NEWLINE(4)
SPC(6)
RBRACES(15)
NEWLINE(4)
NEWLINE(4)
SPC(6)
ELSE(54)
SPC(6)
IF(53)
SPC(6)
IDENTIFIER(3)
SPC(6)
EQUAL(27)
SPC(6)
IMMEDIATE(2)
SPC(6)
LBRACES(14)
NEWLINE(4)
SPC(6)
IDENTIFIER(3)
LPAREN(10)
IDENTIFIER(3)
LBRACKETS(12)
IMMEDIATE(2)
RBRACKETS(13)
RPAREN(11)
NEWLINE(4)
SPC(6)
IDENTIFIER(3)
LPAREN(10)
IDENTIFIER(3)
LBRACKETS(12)
IMMEDIATE(2)
RBRACKETS(13)
RPAREN(11)
NEWLINE(4)
SPC(6)
RBRACES(15)
NEWLINE(4)
NEWLINE(4)
SPC(6)
ELSE(54)
SPC(6)
LBRACES(14)
NEWLINE(4)
SPC(6)
IDENTIFIER(3)
LPAREN(10)
IDENTIFIER(3)
COMMA(8)
SPC(6)
IDENTIFIER(3)
IDENTIFIER(3)
RPAREN(11)
NEWLINE(4)
SPC(6)
RBRACES(15)
NEWLINE(4)
NEWLINE(4)
SPC(6)
RETURN(57)
SPC(6)
IMMEDIATE(2)
NEWLINE(4)
RBRACES(15)
EOI(1)
